### PR TITLE
DTSPO-4471 - add premium servicebus

### PR DESCRIPTION
**Pull request by endakelly**
### JIRA link (if applicable) ###
https://tools.hmcts.net/browse/DTSPO-4471

### Change description ###
Adding premium tier service bus to bulk-scan

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -61,3 +61,7 @@ variable "wafFileUploadLimit" {
 }
 
 variable "aks_subscription_id" {}
+
+variable "sku" {
```
Comment (timja) [inline variables.tf:65]: better to just hardcode it, unless there's a need to have a different value in different environments

File: queue-demo-premium.tf [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  value = "${module.processed-envelopes-queue-premium.primary_listen_connection_string}"
+}
+
+output "processed_envelopes_queue_primary_send_connection_string_premium" {
+  value = "${module.processed-envelopes-queue-premium.primary_send_connection_string}"
+}
+
+output "envelopes_queue_max_delivery_count_premium" {
+  value = "${var.envelope_queue_max_delivery_count}"
+}
```
Comment (timja) [inline queue-demo-premium.tf:157]: are these needed?

File: queue-demo-premium.tf [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+resource "azurerm_key_vault_secret" "processed_envelopes_queue_send_conn_str_premium" {
+  name      = "processed-envelopes-queue-send-connection-string-premium"
+  value     = "${module.processed-envelopes-queue-premium.primary_send_connection_string}"
+  key_vault_id = "${module.vault.key_vault_id}"
+}
+
+resource "azurerm_key_vault_secret" "processed_envelopes_queue_listen_conn_str_premium" {
+  name      = "processed-envelopes-queue-listen-connection-string-premium"
+  value     = "${module.processed-envelopes-queue-premium.primary_listen_connection_string}"
```
Comment (timja) [inline queue-demo-premium.tf:110]: can you fix the 'interpolation only string' deprecation warnings throughout in your change please

ideally filing a separate PR before hand to fix any other issues in the repo

```suggestion
  value     = module.processed-envelopes-queue-premium.primary_listen_connection_string
```

Comment (endakelly) [reply queue-demo-premium.tf:110]: See https://github.com/hmcts/bulk-scan-shared-infrastructure/pull/255

File: queue-demo-premium.tf (hunk 2 above)
Comment (endakelly) [reply queue-demo-premium.tf:157]: I can't imagine why they would be as the connection strings in particular get added to keyvault as secrets anyway

**Review by timja — APPROVED** [review]

**Pull request merged**